framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -248,7 +421,7 @@ class BlockSynchronizationMechanism {
 	async isValidFor(receivedBlock) {
 		// 2. Step: Check whether current chain justifies triggering the block synchronization mechanism
 		const finalizedBlock = await this.storage.entities.Block.getOne({
-			height_eq: this.bft.finalizedHeight,
+			height_eql: this.bft.finalizedHeight,
```

Comment:
I think we can do this?
```suggestion
			height: this.bft.finalizedHeight,
```